### PR TITLE
Standardise the error message for missing results_dir

### PR DESCRIPTION
**Pull request by WilliamMcCumstie**
The distinction between a job not reporting its `results_dir` and the directory not existing as been removed.

Moving forward, all successfully submitted jobs will have a `results_dir`. A generic "no results found" is emitted when the directory does not exist. The same error is emitted for failed submissions.

### CODE REVIEW

**Comment by benarmston** [comment]
Rebased on `master`.

File: lib/flight_job/commands/list_job_results.rb [hunk 1]
```diff
@@ -37,7 +37,13 @@ class ListJobResults < Command
 
       def run
         job = load_job(args.first)
-        assert_results_dir_exists(job)
+
+        # Skip the assertion check when hard wrapping and the directory exists
+        # This causes to command to run on empty directories
+        if hard_wrap? && Dir.exists?(job.results_dir)
```
Comment (benarmston) [inline lib/flight_job/commands/list_job_results.rb:43]: I don't think this is doing what you want it to.

Comment (benarmston) [reply lib/flight_job/commands/list_job_results.rb:43]: Also `hard_wrap` is a confusing name.  I'm not sure what part of passing additional options to the `ls` command is hard or why its wrapped.  Or is hard wrap the opposite of a soft wrap?

Can we get a better name for this please?

Also let's rename `HARD_ERROR` to `LS_ERROR`.

File: lib/flight_job/command.rb [hunk 2]
```diff
@@ -203,8 +203,15 @@ def load_job(id)
     def assert_results_dir_exists(job)
       # NOTE: Jobs created with old versions of flight-job will not have a
       # results directory.
-      unless job.results_dir
-        raise MissingError, "The job did not report its results directory"
+      if !(job.results_dir && Dir.exists?(job.results_dir))
+        case job.state
+        when 'PENDING'
+          raise MissingError, 'Your job has not started, please try again latter...'
```
Comment (benarmston) [inline lib/flight_job/command.rb:209]: `latter` should be spelled `later`.  Here and elsewhere please.

File: lib/flight_job/command.rb [hunk 3]
```diff
@@ -203,8 +203,15 @@ def load_job(id)
...
+        when *Job::RUNNING_STATES
+          raise MissingError, 'No job results found, please try again latter...'
+        else
+          raise MissingError, 'No job results found.'
+        end
```
Comment (benarmston) [inline lib/flight_job/command.rb:214]: These error messages can also be printed directly by the `ListJobResultsCommand#run` method.  When printed by the `run` method they are coloured.  When raised here they aren't.  Can we get these coloured too please?

File: lib/flight_job/commands/list_job_results.rb (hunk 1 above)
Comment (benarmston) [reply lib/flight_job/commands/list_job_results.rb:43]: Also, why do we want this branch.  `assert_results_dir_exists` doesn't check for empty directories.

**Review by benarmston — CHANGES_REQUESTED** [review]
This doesn't work.  I still get the raw `ls` error aboiut the directory not existing.

File: lib/flight_job/command.rb (hunk 2 above)
Comment (WilliamMcCumstie) [reply lib/flight_job/command.rb]: 57a3f08de3ee67b0da9e0992f505c22fad679b9b

File: lib/flight_job/commands/list_job_results.rb (hunk 1 above)
Comment (WilliamMcCumstie) [reply lib/flight_job/commands/list_job_results.rb]: I've added the assertion check back here: 57a3f08de3ee67b0da9e0992f505c22fad679b9b.

I originally removed the check when hard wrapping `ls` to preserve the original error. However this isn't required.

Comment (WilliamMcCumstie) [reply lib/flight_job/commands/list_job_results.rb]: Renamed `LS_ERROR`: 583e9de2742b3684a28248ece69a625c1ab8674f

File: lib/flight_job/command.rb (hunk 3 above)
Comment (WilliamMcCumstie) [reply lib/flight_job/command.rb]: I am not entirely sure why the colourisation isn't working. The top level error handler should be applying the default colours. This _may_ be an issue with your `TERM` setting as I can not reproduce it.

---


Regardless, I have collapsed the two sets of warnings/errors into a single method:
7f2ce35fef7732fdf6542cf7f9233dbfdf5113af

**Comment by benarmston** [comment]
LGTM.

**Pull request merged**